### ISSUE TITLE
Specify channels in recipe.yaml

### ISSUE THREAD

**Issue by pavelzw** [prefix-dev/rattler-build#262]
AFAIK, the only way to specify a custom channel for rattler-build is in the CLI using `-c <custom-channel>`. 

It would be nice to be able to also specify this in the recipe.yaml.

**Comment by wolfv** [comment]
Yeah, I guess we're not sure if `recipe.yaml` is the right place for this. 
We could do it via `variant_config.yaml`, maybe?

**Comment by wolfv** [comment]
Or some other global config file (like `.condarc` does).

**Comment by pavelzw** [comment]
In variant_config.yaml sounds good 👍🏻

**Comment by jacksonwb** [comment]
Or in the `extra` section of the `meta.yaml` like `conda-lock` does.

**Comment by chebee7i** [comment]
Not sure I'd want this in `.condarc` unless there was a section specifically tailored for `rattler-build`.

I'd also like some way to specify this via config.  We have strictly internal packages that depend on other internal packages, so every recipe we build needs to be built with a custom channel.

**Comment by marvinvanaalst** [comment]
How about specific channels for specific packages?

In `pixi` I can specify exactly where to search for a specific package

```toml
[tool.pixi.dependencies]
blast = {channel = "bioconda", version = ">=2.15.0"}
```

but when generating a `recipe.yaml` for this I need to add the `-c bioconda` flag to `rattler-build` and there is no information in the yaml file about where to get the dependency from.

```yaml
requirements:
  run:
  - blast >=2.15.0
```

Probably it's just my internal need for symmetry, but are there maybe also cases where the explicit channel in `pixi` will lead to a different dependency than in `rattler-build`?

Or am I just missing some other way of doing this?

**Comment by geoHeil** [comment]
I want to use a corporate artifactory hosing my mirror of conda-forge. I fear this (missing) feature discussed here is also relevant for this usecase - what options d oI have to replicate what pixi has for mirrors:

```toml
[mirrors]
"https://conda.anaconda.org/" = [
    "company mirror"
```

over to rattler?

to source the packages from the mirror and then push to my corporate artifactory?

So far I just see

- https://github.com/prefix-dev/rattler-build/issues/1304 
- https://github.com/prefix-dev/rattler-build/issues/834

**Comment by pavelzw** [comment]
so far you need to run `rattler-build build -c https://my-company.mirror/conda-forge ...` and then it should work

**Comment by geoHeil** [comment]
What about the pypi mirror? As this will only change the one for CONDA forge and fail in case of mixed environments

**Comment by geoHeil** [comment]
So for such mixed environments you would redirect to pixi build? Or suggest never cause such a mixend environment?

Especially with pytorch https://pixi.sh/latest/python/pytorch/ now going all in on pypi I would imagine having an answer to this wich is easy to implement would be rather neat.

And so far I was often facing issues with the new conda-forge-based version of pytorch after 2.5

**Comment by pavelzw** [comment]
we decided to use the channels specifiable in the new config file (still needs some docs, will do that soon)
